search-commons/src/main/java/no/unit/nva/search2/common/SearchQuery.java

```diff
@@ -177,13 +182,24 @@ protected Stream<Entry<K, QueryBuilder>> builderStreamDefaultQuery(K key) {
     @Override
     public Stream<QueryContentWrapper> assemble() {
         var builder = builderDefaultSearchSource();
-
         handleFetchSource(builder);
+
+        var requests = Integer.valueOf(includeAggregation() ? 2 : 1);
+        var contentWrappers = new ArrayList<QueryContentWrapper>(requests);
+        if (requests.equals(2)) {
+            builder.size(0);
```

Comment:
Can you make the `0` a constant with a name reflecting your comment below, like

`private static final int ZERO_RESULTS_AGGREGATION_ONLY = 0`